packages/api/src/codegen/languages/typescript/index.ts

```diff
@@ -458,25 +392,117 @@ sdk.server('https://eu.api.example.com/v14');`),
     });
 
     sourceFile.addExportAssignment({
-      // Because CJS targets have `createSDK` exported with `module.exports`, but the TS type side
-      // of things to work right we need to set this as `export =`. Thankfully `ts-morph` will
-      // handle this accordingly and still create our JS file with `module.exports` and not
-      // `export =` -- only TS types will have this export style.
-      isExportEquals: this.compilerTarget === 'cjs' && this.outputJS,
+      // Because we're exporting `createSDK` as an IIFE constant we need to have it exported as
+      // `export default createSDK`. `addExportAssignment` by default wants it exported as
+      // `export = createSDK`, which will throw TS errors because we may also be exporting types in
+      // the `./types.ts` file.
+      isExportEquals: false,
       expression: 'createSDK',
     });
 
     return sourceFile;
   }
 
+  /**
+   * Create the `tsconfig.json` file that will allow this SDK to be compiled for use.
+   *
+   */
+  createTSConfig() {
+    const sourceFile = this.project.createSourceFile('tsconfig.json', '');
+
+    const config: TsConfigJson = {
+      compilerOptions: {
+        checkJs: true,
+        esModuleInterop: true,
+        module: 'NodeNext',
+        resolveJsonModule: true,
+      },
+      include: ['./src/**/*'],
+      exclude: ['dist'],
+    };
+
+    sourceFile.addStatements(JSON.stringify(config, null, 2));
+
+    return sourceFile;
+  }
+
+  /**
+   * Create the `package.json` file that will ultimately make this SDK available to use.
+   *
+   */
+  createPackageJSON() {
+    const sourceFile = this.project.createSourceFile('package.json', '');
+
+    const hasTypes = !!Object.keys(this.schemas).length;
+
+    const info = this.spec.getDefinition().info;
+    let pkgVersion = semver.coerce(info.version);
+    if (!pkgVersion) {
+      // If the version that's in `info.version` isn't compatible with semver NPM won't be able to
+      // handle it properly so we need to fallback to something it can.
+      pkgVersion = semver.coerce('0.0.0') as SemVer;
+    }
+
+    const tsupOptions: Options = {
+      cjsInterop: true,
+      clean: true,
+      dts: true,
+      entry: [
+        './src/index.ts',
+        // If this SDK has schemas and generated types then we should also export those too so
+        // they're available to use.
+        hasTypes ? './src/types.ts' : '',
+      ].filter(Boolean),
+      // TODO: figure this out
+      // external: ['@readme/api-core'],
```

Comment:
I'm not really clear on what `external` is for but in my limited testing I don't think we need it?

```suggestion
```

Reply:
yeah i kept this as a TODO to myself because i was wondering if it made sense to bundle `@readme/api-core` in the code-gen'd SDK or to keep it as an npm dep. think we can safely remove this